crates/noirc_evaluator/src/ssa/acir_gen.rs

```diff
@@ -1272,10 +1326,52 @@ fn bound_check_with_offset(
         bits < FieldElement::max_num_bits(),
         "range check with bit size of the prime field is not implemented yet"
     );
-    let aof = add(a, FieldElement::one(), offset);
+
+    let mut aof = add(a, FieldElement::one(), offset);
+
+    if b.is_const() && b.q_c.fits_in_u128() {
+        let f = if *offset == Expression::one() {
+            aof = a.clone();
+            assert!(b.q_c.to_u128() >= 1);
+            b.q_c.to_u128() - 1
+        } else {
+            b.q_c.to_u128()
+        };
+
+        if f < 3 {
+            match f {
+                0 => evaluator.gates.push(Gate::Arithmetic(aof)),
+                1 => {
+                    let expr = boolean_expr(&aof, evaluator);
+                    evaluator.gates.push(Gate::Arithmetic(expr))
+                }
+                2 => {
+                    let y = InternalVar::expression_to_witness(
+                        boolean_expr(&aof, evaluator),
+                        evaluator,
+                    );
+                    let two = FieldElement::from(2_i128);
+                    let y_expr = from_witness(y);
+                    let eee = subtract(&mul_with_witness(evaluator, &aof, &y_expr), two, &y_expr);
+                    evaluator.gates.push(Gate::Arithmetic(eee));
+                }
+                _ => unreachable!(),
+            }
+            return;
+        }
+        let bit_size = bit_size_u128(f);
+        if bit_size < 128 {
+            let r = (1_u128 << bit_size) - f - 1;
+            assert!(bits + bit_size < FieldElement::max_num_bits()); //we need to ensure a+r does not overflow
+            let aor = add(&aof, FieldElement::from(r), &Expression::one());
+            let witness = InternalVar::expression_to_witness(aor, evaluator);
+            range_constraint(witness, bit_size, evaluator).unwrap();
```

Comment:
try_range_constraint could be used here as well

Reply:
Done